src/main/io/serial.c

```diff
@@ -421,3 +423,45 @@ void evaluateOtherData(serialPort_t *serialPort, uint8_t receivedChar)
         systemResetToBootloader();
     }
 }
+
+/*
+A low-level interface for implementing serial passthrough. This is used for
+example by gps to do serial passthrough while also consuming GPS data and
+updating the display. It is used below by the high-level `serialPassthrough()`
+function to move data between two serial ports.
+
+Read data from `left` and writes it to `right`
+Returns data (if any) to caller for further processing.
+*/
+uint8_t serialPassthroughStep(serialPort_t *left, serialPort_t *right)
+{
+    uint8_t c = 0;
+    if (serialRxBytesWaiting(left)) {
+        c = serialRead(left);
+        serialWrite(right, c);
+    }
+    return c;
+}
+
+/*
+A high-level serial passthrough implementation. Used by cli to start an
+arbitrary serial passthrough "proxy". Manages LEDs while using the low-level
+`serialPassthroughStep()` to shuffle data both ways.
+*/
+void serialPassthrough(serialPort_t *left, serialPort_t *right)
```

Comment:
Nice suggestion. This cleans up the code a bit. I could not do proper LED management the way I had it.